packages/next-upgrade/index.ts

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env node
+
+import prompts from 'prompts'
+import fs from 'fs'
+import { execSync } from 'child_process'
+import path from 'path'
+import { compareVersions } from 'compare-versions'
+
+type Version = 'latest' | 'rc' | 'canary'
+
+interface Response {
+  version: Version
+}
+
+async function run(): Promise<void> {
+  let version: Version
+
+  const specifiedVersion = process.argv[2]?.replace('@', '')
+  if (['latest', 'rc', 'canary'].includes(specifiedVersion)) {
+    // Shortcut
+    version = specifiedVersion as Version
+  } else {
+    const response: Response = await prompts({
+      type: 'select',
+      name: 'version',
+      message: 'What version of Next.js do you want to upgrade to?',
+      choices: [
+        { title: 'Latest Stable', value: 'latest' },
+        { title: 'Latest Release Candidate', value: 'rc' },
+        { title: 'Canary', value: 'canary' },
```

Comment:
Thoughts on the word choices? Should we just say "Stable" and "Release Candidate" instead of "Latest Stable" and "Latest Release Candidate"? @elsigh